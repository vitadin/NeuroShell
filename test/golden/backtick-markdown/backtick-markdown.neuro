%% Markdown backtick functionality test
%% Verifies that markdown-style code blocks with backticks work correctly

%% Test inline code with single backticks
\set[inline_code="Use `console.log()` for debugging"]
\set[function_ref="Call the `getData()` function"]

%% Test code blocks with triple backticks  
\set[code_block="```\nfunction hello() {\n  console.log('world');\n}\n```"]
\set[bash_block="```bash\nls -la\ngrep pattern file.txt\n```"]

%% Test nested and mixed backticks
\set[nested="Inside `code` we can have ```blocks```"]
\set[complex="Mix `inline`, ```blocks```, and 'quotes'"]

%% Test markdown examples
\set[readme="# README\n\nInstall with `npm install`\n\n```javascript\nconst app = require('app');\n```"]

%% Retrieve all variables
\get[inline_code]
\get[function_ref]
\get[code_block]
\get[bash_block]
\get[nested]
\get[complex]
\get[readme]