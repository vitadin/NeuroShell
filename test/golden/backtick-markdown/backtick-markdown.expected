Setting _echo_command = true
%%> "\\set[inline_code=\"Use `console.log()` for debugging\"]"
Setting inline_code = Use `console.log()` for debugging
%%> "\\set[function_ref=\"Call the `getData()` function\"]"
Setting function_ref = Call the `getData()` function
%%> "\\set[code_block=\"```\\nfunction hello() {\\n  console.log('world');\\n}\\n```\"]"
Setting code_block = ```\nfunction hello() {\n  console.log('world');\n}\n```
%%> "\\set[bash_block=\"```bash\\nls -la\\ngrep pattern file.txt\\n```\"]"
Setting bash_block = ```bash\nls -la\ngrep pattern file.txt\n```
%%> "\\set[nested=\"Inside `code` we can have ```blocks```\"]"
Setting nested = Inside `code` we can have ```blocks```
%%> "\\set[complex=\"Mix `inline`, ```blocks```, and 'quotes'\"]"
Setting complex = Mix `inline`, ```blocks```, and 'quotes'
%%> "\\set[readme=\"# README\\n\\nInstall with `npm install`\\n\\n```javascript\\nconst app = require('app');\\n```\"]"
Setting readme = # README\n\nInstall with `npm install`\n\n```javascript\nconst app = require('app');\n```
%%> "\\get[inline_code]"
inline_code = Use `console.log()` for debugging
%%> "\\get[function_ref]"
function_ref = Call the `getData()` function
%%> "\\get[code_block]"
code_block = ```\nfunction hello() {\n  console.log('world');\n}\n```
%%> "\\get[bash_block]"
bash_block = ```bash\nls -la\ngrep pattern file.txt\n```
%%> "\\get[nested]"
nested = Inside `code` we can have ```blocks```
%%> "\\get[complex]"
complex = Mix `inline`, ```blocks```, and 'quotes'
%%> "\\get[readme]"
readme = # README\n\nInstall with `npm install`\n\n```javascript\nconst app = require('app');\n```